lms/djangoapps/instructor_task/api.py

```diff
@@ -551,12 +553,14 @@ def regenerate_certificates(request, course_key, statuses_to_regenerate):
 
     instructor_task = submit_task(request, task_type, task_class, course_key, task_input, task_key)
 
-    CertificateGenerationHistory.objects.create(
-        course_id=course_key,
-        generated_by=request.user,
-        instructor_task=instructor_task,
-        is_regeneration=True
-    )
+    cert_filter_args = {
+        "course_id": course_key,
+        "generated_by": request.user,
+        "instructor_task": instructor_task,
+        "is_regeneration": True
+    }
+
+    create_or_update_certificate_generation_history(**cert_filter_args).save()
```

Comment:
save() method is redundant
```suggestion
    create_or_update_certificate_generation_history(**cert_filter_args)
```

Reply:
Updated!